src/Ladybug_SolarEnvelopeAdvanced.py

```diff
@@ -0,0 +1,160 @@
+        self.buildSunPosList(azimuthAngles, alltitudeAngles)
+        self.getPtsFromClosedCrv(_baseSrf,_gridSize)
+        self.parallelFindPointHeights(self.checkPointList, obstacleCurves)
+    
+    def getFinalPointList(self):
+        for gridPt in self.gridPoints:
+            self.finalPointsList.Add(gridPt.buildFinalPoint())
+        return self.finalPointsList
+    def getPtsFromClosedCrv(self,srf,gridSize):
+        regionMeshPar = Rhino.Geometry.MeshingParameters.Default
+        regionMeshPar.MinimumEdgeLength = regionMeshPar.MaximumEdgeLength = gridSize/2
+        regionMesh = Rhino.Geometry.Mesh.CreateFromBrep(srf, regionMeshPar)[0]
+        vertices = regionMesh.Vertices
+        self.checkPointList = []
+        for item in vertices:
+            self.checkPointList.append(Rhino.Geometry.Point3d(item))
+class GridPt:
+    def __init__(self, point, defaultHeight,mainRef):
+        self.point = point
+        self.defaultHeight = self.height = defaultHeight
+        self.mainRef = mainRef
+        self.isStart = False        
+    #handle all the logic and return the z (height) of the relevant point for one specified obstacle line
+    #if the z value is lower than what we had replace it (because the lowest one is the relevant one)
+    def getPointHeight(self, bLine, singleSun):
+        self.initialHeight = bLine.PointAtEnd.Z
+        _checkPoint = Rhino.Geometry.Point2d(self.point[0],self.point[1])
+        xAdd = - self.mainRef.lineExtention * math.sin(math.radians(singleSun.azimuth));
+        yAdd = - self.mainRef.lineExtention * math.cos(math.radians(singleSun.azimuth)); 
+        point1 = Rhino.Geometry.Point3d( self.point[0],self.point[1],self.initialHeight )
+        point2 = Rhino.Geometry.Point3d( self.point[0] + xAdd,self.point[1] + yAdd,self.initialHeight)
+        _sunLine = Rhino.Geometry.LineCurve(point1,point2)       
+        _intersections = Rhino.Geometry.Intersect.Intersection.CurveCurve(_sunLine, bLine, 0.001, 0.0)
+        if _intersections : 
+            _intersectionPoint = Rhino.Geometry.Point2d(_intersections[0].PointA[0],_intersections[0].PointA[1])
+            dist = (_intersectionPoint - _checkPoint).Length
+            t = math.tan(math.radians(singleSun.alltitude))
+            return dist * t + self.initialHeight
+        else :
+            #sun not relevant so no obstacles to look out for - return the heighest point defined
+            return self.defaultHeight + self.initialHeight
+    
+    def buildFinalPoint(self):
+        self.finalPoint = Rhino.Geometry.Point3d(self.point[0],self.point[1],self.height) 
+        return self.finalPoint
+#class to organize all the data in a single sun object
+#properties for later use, when we'll get more comprehensive data from and epw file - hour, day, month, temperature, radiation
+class SingleSun:
+    def __init__(self, _alltitude, _azimuth):
+        self.alltitude = _alltitude
+        self.azimuth = _azimuth
+
+if _runIt:
```

Comment:
@boris-p here you probably want to make sure that all the other required inputs are also provided. Something like:
```python
if _runIt and _baseSrf and len(_sunVectors)!=0 and (other required inputs):
```